library/src/test/java/com/vinted/coper/CoperImplTest.kt

```diff
@@ -374,25 +372,62 @@ class CoperImplTest {
     @Test
     fun getFragment_twoRequest_sameInstance() {
         runBlocking {
-            val fixture = CoperImpl(activity.supportFragmentManager)
-            val firstFragment = fixture.getFragment()
-            val secondFragment = fixture.getFragment()
+            val fixture = getCoperInstance()
+            val firstFragment = fixture.getFragmentSafely()
+            val secondFragment = fixture.getFragmentSafely()
 
             assertTrue { firstFragment === secondFragment }
         }
     }
 
-    @Test(expected = IllegalStateException::class)
-    fun request_onIoThread_shouldCrash() {
-        val fixture = spy(CoperImpl(activity.supportFragmentManager))
-        fixture.mockGetFragmentWithStub()
-
-        runBlocking(Dispatchers.IO) {
-            executePermissionRequest(
-                coperImplReference = fixture,
-                permissions = listOf("test"),
-                permissionResult = listOf(PermissionChecker.PERMISSION_GRANTED)
+    @Test
+    @Ignore(
+        """
+        Need better solution, because fragment manager could only do transaction on main thread,
+        but when main thread is already in used by test, so fragment transaction queue never starts
+        its work, so if fragment is already in backstack check never be successful
+    """
+    )
```

Comment:
If you have ideas how to make this test to pass, just write it down here

Reply:
Yield? 

But something is wrong with a test. Instead of checking if some method was called, you should check the potential negative impact. For example, you want to avoid multiple `CoperFragment` added, also, you want all requests will succeed even if they was requested in the same time. You should avoid `spy` at any costs. Also, keep in mind that `spy` wraps an instance, but does not change it.

Reply:
As I debugged, spy creates different instance of my object, for some of my examples it created separate instance

Reply:
What the problem is, that I need asynchronously launch 10 coroutines, that one could finish somehow, and it could only finish it when main thread is free, but test is already taken main thread

Reply:
How someone could steal thread context? Do you mean the test is using Looper thread? It sounds strange to me.

Reply:
As I understand, the problem is that `runBlocking {` blocks main thread while test is running and therefore fragment manager can't perform its transactions on the main thread while test is running? 

Maybe inside implementation you could switch coroutine context to use `Main` dispatcher for fragment manager operation? `withContext(Dispatchers.Main) { // add fragment to manager } `

Reply:
Yes this would work, but it would change implementation for test to pass, I didnt wanted that

Reply:
You could have similar solution as you had with `internal val isTestDispatcher = false` and use separate dispatcher for tests. Maybe it would help to simplify your test.

Reply:
I already made changes, it was problem that main looper is paused and need to run all actions added to that looper

Reply:
Yes, that works, but doesn't look like idiomatic testing solution. Usually in such situations dependencies are substituted to test environment specific dependencies.

Reply:
What do you mean, could you elaborate on this. My problem is common as I understood, when posting to main looper it is paused and some those actions do not pass through that looper it needs some push 😅 http://robolectric.org/blog/2019/06/04/paused-looper/